### DISCUSSION TITLE
H prediction is detached in Sydney model

### DISCUSSION THREAD

**Discussion by wkirgsn** [moetomg/magnet-engine#4] [General]
Hey,

I came across this code in the forward function of the H prediction model of the Sydney team:
```py
# Return results
H = savgol_filter(H.detach().to("cpu").numpy(), window_length=7, polyorder=2, axis=1)
H = torch.from_numpy(H).view(batch_size, -1, 1)
```

Is it intended to detach the H prediction here, which cuts off the computational graph at that point?
Gradients wouldn't be computable backwards beyond that point anymore.

Btw, short reminder that the artifacts are still there

Best regards

**Comment by moetomg** [comment]
Hi Wilhelm, 

Thanks you very much for working through carefully. Indeed, these two lines of code do not pertain to the main part of the Sydney core loss model, and they are just for better presentation of data. Since the main purpose of this responsitory is to present data more friendly, we added these two lines of code, which is equivalent to a low pass filter in hardware, eliminating a little spikes at the initial place due to numerical problems.  

If you want to reproduce/train the Sydney model, you should remove it. Otherwise, even keeping the code will not significantly affect the results of the model (less than 1% variation in the core loss prediction, reduce 2-3 glitches in the H prediction).

Best regards,
Tommy
